kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -488,6 +488,110 @@ class ResidualBasedBlockBuilderAndSolver
+                << " Old Stiffness on First Iteration. \n"
+                << "setting mUseOldStiffnessInFirstIteration=false " << std::endl;
+        }
+
+        DofsArrayType FixedDofs;
+        for(auto& dof : BaseType::mDofSet){
+            if(dof.IsFixed()){
+                FixedDofs.push_back(&dof);
+                dof.FreeDof();
+            }
+        }
+
+        //TODO: Here we need to take the vector from other ones because
+        // We cannot create a trilinos vector without a communicator. To be improved!
+        TSystemVectorType dx_prediction(rDx);
+        TSystemVectorType rhs_addition(rb); //we know it is zero here, so we do not need to set it
+
+        // Here we bring back the database to before the prediction,
+        // but we store the prediction increment in dx_prediction.
+        // The goal is that the stiffness is computed with the
+        // converged configuration at the end of the previous step.
+        const auto it_dof_begin = BaseType::mDofSet.begin();
+        #pragma omp parallel for
+        for (int i = 0; i < static_cast<int>(BaseType::mDofSet.size()); ++i) {
+            auto it_dof = it_dof_begin + i;
+            //NOTE: this is initialzed to - the value of dx prediction
+            dx_prediction[it_dof->EquationId()] = -(it_dof->GetSolutionStepValue() - it_dof->GetSolutionStepValue(1));
+        }
+
+        // Use UpdateDatabase to bring back the solution to how it was at the end of the previous step
+        pScheme->Update(rModelPart, BaseType::mDofSet, rA, dx_prediction, rb);
+        const auto it_node_begin = rModelPart.NodesBegin();
+        if (MoveMesh) {
+            #pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i) {
+                auto it_node = it_node_begin + i;
+                noalias(it_node->Coordinates()) = it_node->GetInitialPosition().Coordinates()
+                                                + it_node->FastGetSolutionStepValue(DISPLACEMENT);
+            }
+        }
+
+        this->Build(pScheme, rModelPart, rA, rb);
+
+        // Put back the prediction into the database
+        TSparseSpace::InplaceMult(dx_prediction, -1.0); //change sign to dx_prediction
+        TSparseSpace::UnaliasedAdd(rDx, 1.0, dx_prediction);
+
+        // Use UpdateDatabase to bring back the solution
+        // to where it was taking into account BCs
+        // it is done here so that constraints are correctly taken into account right after
+        pScheme->Update(rModelPart, BaseType::mDofSet, rA, dx_prediction, rb);
+        if (MoveMesh) {
+            #pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i) {
+                auto it_node = it_node_begin + i;
+                noalias(it_node->Coordinates()) = it_node->GetInitialPosition().Coordinates()
+                                                + it_node->FastGetSolutionStepValue(DISPLACEMENT);

```

Comment:
Aside from the hardcoding, I think there is a fct for this in the variableutils

Reply:
Something that could be done, but is a much larger change, is to have the function MoveMesh in the base scheme instead of in the strategy. That way one may reimplement it.

However as i am telling this is a large change (it would require revising all the schemes) so i really would not like doing it as a part of this PR